vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1009 @@
+    size_t                     d2start[2];
+    size_t                     d3count[3];
+    size_t                     d3start[3];
+    size_t                     d4count[4];
+    size_t                     d4start[4];
+    size_t                     d5count[5];
+    size_t                     d5start[5];
+    size_t                     d6count[6];
+    size_t                     d6start[6];
+
+    // validate state file dimensions and coordinate variables
+    check_init_state_file();
+
+    // read state variables
+
+    // allocate memory for variables to be stored
+    cvar = malloc(local_domain.ncells_active * sizeof(*cvar));
+    if (cvar == NULL) {
+        log_err("Memory allocation error in vic_restore().");
+    }
+
+    ivar = malloc(local_domain.ncells_active * sizeof(*ivar));
+    if (ivar == NULL) {
+        log_err("Memory allocation error in vic_restore().");
+    }
+
+    dvar = malloc(local_domain.ncells_active * sizeof(*dvar));
+    if (dvar == NULL) {
+        log_err("Memory allocation error in vic_restore().");
+    }
+
+    fvar = malloc(local_domain.ncells_active * sizeof(*fvar));
+    if (fvar == NULL) {
+        log_err("Memory allocation error in vic_restore().");
+    }
+
+    // initialize starts and counts
+    d2start[0] = 0;
+    d2start[1] = 0;
+    d2count[0] = global_domain.n_ny;
+    d2count[1] = global_domain.n_nx;
+
+    d3start[0] = 0;
+    d3start[1] = 0;
+    d3start[2] = 0;
+    d3count[0] = 1;
+    d3count[1] = global_domain.n_ny;
+    d3count[2] = global_domain.n_nx;
+
+    d4start[0] = 0;
+    d4start[1] = 0;
+    d4start[2] = 0;
+    d4start[3] = 0;
+    d4count[0] = 1;
+    d4count[1] = 1;
+    d4count[2] = global_domain.n_ny;
+    d4count[3] = global_domain.n_nx;
+
+    d5start[0] = 0;
+    d5start[1] = 0;
+    d5start[2] = 0;
+    d5start[3] = 0;
+    d5start[4] = 0;
+    d5count[0] = 1;
+    d5count[1] = 1;
+    d5count[2] = 1;
+    d5count[3] = global_domain.n_ny;
+    d5count[4] = global_domain.n_nx;
+
+    d6start[0] = 0;
+    d6start[1] = 0;
+    d6start[2] = 0;
+    d6start[3] = 0;
+    d6start[4] = 0;
+    d6start[5] = 0;
+    d6count[0] = 1;
+    d6count[1] = 1;
+    d6count[2] = 1;
+    d6count[3] = 1;
+    d6count[4] = global_domain.n_ny;
+    d6count[5] = global_domain.n_nx;
+
+    // total soil moisture
+    for (m = 0; m < options.NVEGTYPES; m++) {
+        d5start[0] = m;
+        for (k = 0; k < options.SNOW_BAND; k++) {
+            d5start[1] = k;
+            for (j = 0; j < options.Nlayer; j++) {
+                d5start[2] = j;
+                get_scatter_nc_field_double(filenames.init_state,
+                                            "Soil_moisture",
+                                            d5start, d5count, dvar);
+                for (i = 0; i < local_domain.ncells_active; i++) {
+                    v = veg_con_map[i].vidx[m];
+                    if (v >= 0) {
+                        all_vars[i].cell[v][k].layer[j].moist = dvar[i];
```

Comment:
I think if Ted follows the logic in the other places where `get_scatter` is used, than this should work just fine. The whole point is that at this level the mapping should be invisible.